app/questionnaire_state/state_item.py

```diff
@@ -38,3 +38,10 @@ def get_errors(self):
 
         logger.debug("Item errors list is %s", errors)
         return errors
+
+    def find_state_item(self, schema_item):
+        for child in self.children:
+            if child.schema_item == schema_item:
+                return child
+            else:
+                return child.find_state_item(schema_item)
```

Comment:
I've made this more explicit so that it returns None if schema item is not found.